src/map/skill.c

```diff
@@ -14423,9 +14425,10 @@ static int skill_unit_onplace_timer(struct skill_unit *src, struct block_list *b
 		 * 3rd stuff
 		 **/
 		case UNT_POISONSMOKE:
-			if (battle->check_target(ss,bl,BCT_ENEMY) > 0 && !(tsc != NULL && tsc->data[sg->val2] != NULL)
+			if (battle->check_target(ss, bl, BCT_ENEMY) > 0 && !(tsc != NULL && tsc->data[sg->val2] != NULL)
```

Comment:
I noticed that the effect of those poisons are also implemented in the item scripts, I was thinking about making a `SkillScript` field in item db so we could move the time calculation/etc to the item itself instead of hardcoding the poison logic.

But this condition makes things somewhat complicated:
```
(tsc != NULL && tsc->data[sg->val2] != NULL)
```

I would need to keep it hardcoded, or pass a lot of info to the item script... so I went with the easy approach of just using SkillData2.

If you want to see the initial implementation, I will keep this draft in my own repo for now: https://github.com/guilherme-gm/Hercules/pull/5